### PR TITLE
feat(trailhead): update comms link to go directly to basket

### PR DESCRIPTION
**Pull request by vbudhram**
Fixes https://github.com/mozilla/fxa/issues/962

This patch updates the FxA communication panel to link directly to basket. It also appends the user's email address so that basket can look the user up. It uses the `https://www-dev.allizom.org/newsletter/existing/?email=a%40a.com` url.

This url is configurable for different environments ref https://github.com/mozilla/fxa/blob/baea0b5a7a67e21cfec40b73736c3dde803ee170/packages/fxa-content-server/server/lib/configuration.js#L369.

- [x] Add icon to panel
- [x] Add tests
- [x] Determine the email preferences url that will be linked to

Screen Shot 2019-05-23 at 4 21 35 PM

cc @pmac

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
### Desktop

Screenshot 2019-05-24 at 12 30 58

### Mobile
Screenshot 2019-05-24 at 12 30 52

### RTL
Screenshot 2019-05-24 at 12 30 40

@mozilla/fxa-devs - r?

We'll still need to confirm the URL with @pmac, but we can update that using configuration easily enough.

@pmac - one final question, with this link, will users of all locales be able to opt-in to newsletters? Right now we only show the pane for users [of these locales](https://github.com/mozilla/fxa/blob/04df8cd9417108f6fc03dbed398857e7b6740c62/packages/fxa-content-server/app/scripts/lib/experiments/grouping-rules/communication-prefs.js#L14)

**Comment by shane-tomlinson** [comment]
Note to @jrgm and @jbuck - once this lands, we'll no longer need the basket proxy. 🎆🎆🎆🎆🎆🎆🎆🎆

**Comment by pmac** [comment]
> We'll still need to confirm the URL with @pmac, but we can update that using configuration easily enough.

I got the Oauth flow working last night against stable 🎉 So the Oauth dance is viable for us and should work. There are still questions flying around on whether we can legally use it since it requires auth, but if we do go with the Oauth flow then the links will be in the form `https://basket.mozilla.org/fxa/?email=user%40example.com`.
 
> @pmac - one final question, with this link, will users of all locales be able to opt-in to newsletters? Right now we only show the pane for users [of these locales](https://github.com/mozilla/fxa/blob/04df8cd9417108f6fc03dbed398857e7b6740c62/packages/fxa-content-server/app/scripts/lib/experiments/grouping-rules/communication-prefs.js#L14)

That is a tricky question. Technically all locales can opt-in, but the email preferences page is only translated into roughly that same list of locales, and the newsletters are translated and sent often in even fewer. But a lot of people might choose an available language that isn't their primary choice, so I hesitate to suggest limiting access...

**Comment by pmac** [comment]
I've gotten confirmation that we're going with the Oauth thing. So the URL for email preferences is what I said above:

```
https://basket.mozilla.org/fxa/?email=user%40example.com
```

I have this deployed to basket-stage and working against FxA stable. e.g. `https://basket.allizom.org/fxa/?email=user%40example.com`. The `email` parameter in the URL is optional but provides a better UX.

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - r?

**Comment by clouserw** [comment]
@dannycoates can you review this so we can get it in train 138?

File: packages/fxa-content-server/server/lib/configuration.js [hunk 1]
```diff
@@ -46,24 +46,29 @@ const conf = module.exports = convict({
...
     },
     api_url: {
       default: 'http://127.0.0.1:10140',
-      doc: 'Url for the Basket API server',
+      doc: 'DEPRECATED - Url for the Basket API server',
+      format: String
+    },
+    email_preferences_url: {
+      default: 'https://basket.allizom.org/fxa/',
+      doc: 'DEPRECATED - Url for the Basket proxy server',
```
Comment (pmac) [inline packages/fxa-content-server/server/lib/configuration.js]: @shane-tomlinson this doc string isn't right. And the default URL is for basket staging. This is known right? The prod URL will be different? Perhaps the default should be basket prod?

Comment (shane-tomlinson) [reply packages/fxa-content-server/server/lib/configuration.js]: You are right, and this was intentional. We kinda/sorta go back and forth about whether to put production values as default, or development values. We are settling on development values. I'll open a PR against the cloudops-deployment repo with the correct URL for prod.

**Pull request merged**